L1Trigger/L1THGCal/src/veryfrontend/HGCalVFECompressionImpl.cc

```diff
@@ -1,5 +1,88 @@
 #include "L1Trigger/L1THGCal/interface/veryfrontend/HGCalVFECompressionImpl.h"
 
 HGCalVFECompressionImpl::
-HGCalVFECompressionImpl(const edm::ParameterSet& conf)
-{}
+HGCalVFECompressionImpl(const edm::ParameterSet& conf):
+    exponentBits_(conf.getParameter<uint32_t>("exponentBits")),
+    mantissaBits_(conf.getParameter<uint32_t>("mantissaBits")),
+    rounding_(conf.getParameter<bool>("rounding"))
+{
+  saturationCode_ = (1 << (exponentBits_ + mantissaBits_)) - 1;
+  saturationValue_ = 0xffffffff;
+  if (((1 << exponentBits_) + mantissaBits_ - 1) < 32) {
+    saturationValue_ = (1 << ((1 << exponentBits_) + mantissaBits_ - 1)) - 1;
+  }
+  else {
+    throw cms::Exception("CodespaceCannotFit")
+      << "The code space cannot fit into the unsigned 32-bit space.\n";
```

Comment:
I would rather invert the logic and simplify the code, like:
```cpp
  if ((1 << exponentBits_) + mantissaBits_  > 32) {
    throw cms::Exception("CodespaceCannotFit")
      << "The code space cannot fit into the unsigned 32-bit space.\n";
  }
  saturationCode_ = (1 << (exponentBits_ + mantissaBits_)) - 1;
  saturationValue_ = (1 << ((1 << exponentBits_) + mantissaBits_ - 1)) - 1;
```

Reply:
Done this!